haystack/nodes/retriever/link_content.py

```diff
@@ -177,24 +222,60 @@ def run_batch(
 
         return {"documents": results}, "output_1"
 
-    def _get_response(self, url: str, timeout: Optional[int]) -> requests.Response:
+    def _get_response(self, url: str, timeout: Optional[int] = None) -> requests.Response:
         """
         Fetches content from a URL. Returns a response object.
         :param url: The URL to fetch content from.
         :param timeout: The timeout in seconds.
         :return: A response object.
         """
+
+        @retry(
+            # we want to reraise the exception if we fail after the last self.retry_attempts
+            # then we can catch it in the outer try/except block, see below
+            reraise=True,
+            stop=stop_after_attempt(self.retry_attempts),
+            wait=wait_exponential(multiplier=1, min=2, max=10),
+            retry=(retry_if_exception_type((HTTPError, requests.RequestException))),
+            after=self._switch_user_agent,
```

Comment:
Let's add a small comment here: tenacity's naming is not great, I assumed `after` is run after every request, successful or not.
```suggestion
            # This method is invoked only after failed requests
            after=self._switch_user_agent,
```